crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1307,6 +1307,48 @@ impl<'src> Lexer<'src> {
         }
     }
 
+    /// Re-lex the current token in the context of a logical line.
+    ///
+    /// Returns a boolean indicating that whether the new current token is different than the
+    /// previous current token. This also means that the current position of the lexer has changed
+    /// and the caller is responsible for updating it's state accordingly.
+    ///
+    /// This method is a no-op if the lexer isn't in a parenthesized context.
+    pub(crate) fn re_lex_logical_token(&mut self) -> bool {
```

Comment:
What's your reasoning for returning a `bool` over the re-lexed kind of the token?

Reply:
What I meant was to return the re-lexed token kind or the current token kind if no re-lexing happens, so that it's just a `TokenKind` (and in line with `lex_token`). But if all we need is a bool, than that's fine.